colorscad.sh

```diff
@@ -1,11 +1,11 @@
 #!/usr/bin/env bash
 
```

Comment:
We could enable the "unofficial bash strict mode" perhaps:
```suggestion
set -o errexit -o errtrace -o nounset -o pipefail
trap 'echo "Error $? at $0:$LINENO" >&2' ERR
```
That would hopefully allow removing the various `|| exit 1` too (and those need a test too, first).
But let's maybe do all of that in a followup PR and first get this one in, along with adding shellcheck to the CI.

Reply:
if everything works this way that would be quite nice, but yes, lets postpone that.

Reply:
> set -o errexit -o errtrace -o nounset -o pipefail
> trap 'echo "Error $? at $0:$LINENO" >&2' ERR

just setting this breaks tests, is assume there are some commands that will fail as intended and we would need to build a safeguard around that :/

Reply:
Yes, it's not a trivial switch unfortunately. There are some gotchas too where `|| true` is indeed needed, not just things like `grep` but also `(( expressions ))` that evaluate to 0.